pkg/virtctl/vnc/vnc.go

```diff
@@ -164,76 +168,25 @@ func (o *VNC) Run(cmd *cobra.Command, args []string) error {
 		listenResChan <- err
 	}()
 
-	// execute VNC
-	go func() {
+	vncoptions := kubecli.VNCOptions{
+		Port: ln.Addr().(*net.TCPAddr).Port,
+	}
+	//port := ln.Addr().(*net.TCPAddr).Port
+
+	if onlyProxy {
 		defer close(doneChan)
-		port := ln.Addr().(*net.TCPAddr).Port
-		args := []string{}
-
-		vncBin := ""
-		osType := runtime.GOOS
-		switch osType {
-		case "darwin":
-			if matches, err := filepath.Glob(TIGER_VNC_PATTERN); err == nil && len(matches) > 0 {
-				// Always use the latest version
-				vncBin = matches[len(matches)-1]
-				args = tigerVncArgs(port)
-			} else if err == filepath.ErrBadPattern {
-				viewResChan <- err
-				return
-			} else if _, err := os.Stat(CHICKEN_VNC); err == nil {
-				vncBin = CHICKEN_VNC
-				args = chickenVncArgs(port)
-			} else if !os.IsNotExist(err) {
-				viewResChan <- err
-				return
-			} else if _, err := os.Stat(REAL_VNC); err == nil {
-				vncBin = REAL_VNC
-				args = realVncArgs(port)
-			} else if !os.IsNotExist(err) {
-				viewResChan <- err
-				return
-			} else if _, err := exec.LookPath(REMOTE_VIEWER); err == nil {
-				// fall back to user supplied script/binary in path
-				vncBin = REMOTE_VIEWER
-				args = remoteViewerArgs(port)
-			} else if !os.IsNotExist(err) {
-				viewResChan <- err
-				return
-			}
-		case "linux", "windows":
-			_, err := exec.LookPath(REMOTE_VIEWER)
-			if exec.ErrNotFound == err {
-				viewResChan <- fmt.Errorf("could not find the remote-viewer binary in $PATH")
-				return
-			} else if err != nil {
-				viewResChan <- err
-				return
-			}
-			vncBin = REMOTE_VIEWER
-			args = remoteViewerArgs(port)
-		default:
-			viewResChan <- fmt.Errorf("virtctl does not support VNC on %v", osType)
-			return
+		optionString, err := json.Marshal(vncoptions)
+		if err != nil {
+			return fmt.Errorf("Error encountered: %s", err.Error())
```

Comment:
This error string does not contain any useful information, so it is enough to just return the `err`.

Reply:
It's similar to other error msg in the file, not changing anything here in this PR